### DISCUSSION TITLE
About captain-definition file

### DISCUSSION THREAD

**Discussion by nadveos** [caprover/caprover#1697] [Q&A]
Hi everyone.
How can add a captain-definition file in a Flutter Web Project. Recently i upload my buil to repo but i dont know what write in captain file to build app in my cap rover server. Thanks in advance

**Comment by maietta** [comment]
You can create a `Dockerfile` should look something like this:

```
FROM google/dart

WORKDIR /app

COPY ./pubspec.yaml ./pubspec.lock ./
RUN flutter pub get

COPY . .
RUN flutter build web

EXPOSE 8080

CMD ["dart", "bin/server.dart"
```

You can then deploy this along with your code in a deploy.tar file. Modify the Dockerfile to point to your project source directory and change the port as needed. You'll need to also configure your blank app in CapRover to also use this port defined in your Dockerfile.

You'll also want to add a `captain-definition` file pointing to your Dockerfile.

```
 {
  "schemaVersion": 2,
  "dockerfilePath": "./Dockerfile"
 }
```

**Reply by nadveos** [reply]
Ok, i try this. Thanks
